hibernate-core/src/main/java/org/hibernate/engine/config/internal/ConfigurationServiceImpl.java

```diff
@@ -67,14 +70,14 @@ public <T> T getSetting(String name, Converter<T> converter, T defaultValue) {
 	}
 
 	@Override
-	public <T> T getSetting(String name, Class<T> expected, T defaultValue) {
+	public <T> @Nullable T getSetting(String name, Class<T> expected, @Nullable T defaultValue) {
```

Comment:
I forgot to annotate this method, i gather they are also required (whichever ones we end up putting) ?